py/dml/c_backend.py

```diff
@@ -628,16 +628,12 @@ def generate_attributes(initcode, node, port=None,
             generate_attributes(initcode, child, port, dimsizes, prefix,
                                 loopvars)
     elif node.objtype in {'device', 'bank', 'port'}:
-        assert prefix == ''
-        assert dimsizes == ()
-        assert loopvars == ()
```

Comment:
Shouldn't we keep the asserts for `device`?

Reply:
they are true for `device`, but trivially so, and we no longer make use of it. The new code instead clears them explicitly (well, if you count "omit optional arg" as something explicit).